webapp/lib/view.dart

```diff
@@ -937,127 +937,266 @@ class ResponseListView extends BaseView {
   }
 }
 
-enum FormGroupTypes {
-  TEXT,
-  CHECKBOX,
-  DATE
-}
-
 class ProjectConfigurationView extends BaseView{
-  DivElement configurationViewElement;
+  DivElement _configurationViewElement;
   FormElement _projectConfigurationForm;
+  Map formData;
+  List<String> additionalProjectLanguages;
 
-  ProjectConfigurationView() {
-    configurationViewElement = new DivElement()
+  ProjectConfigurationView(this.formData, this.additionalProjectLanguages) {
```

Comment:
I see you're making changes to `formData` as part of user actions. This means that if the user leaves this page and then comes back to it, the changes will persist (ie. the user doesn't need to press the save button for the changes to appear in the UI).

What we should do is instead make a deep copy of the map in the constructor and locally store that one.